<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!--Launch file for sim-->

  <!-- Arguments -->
  <arg name="use_mock_angle_finder" default="false"/>

  <!-- Nodes -->

  <!-- Uncomment for Nav Channel mock angle finder, and comment out fake_angle_finder -->
  <node name="mock_angle_finder_speed_challenge" pkg="prop_mapper" type="mock_angle_finder_speed_challenge" output="screen">
    <rosparam file="$(find prop_mapper)/config/sim_params.yaml" command="load" />
  </node>

  <!--node name="mock_angle_finder_nav_channel" pkg="prop_mapper" type="mock_angle_finder_nav_channel" output="screen">
    <rosparam file="$(find prop_mapper)/config/sim_params.yaml" command="load" />
  </node-->

  <!--node name="fake_angle_finder" pkg="prop_mapper" type="fake_angle_finder" output="screen" if="$(arg use_mock_angle_finder)">
    <rosparam file="$(find prop_mapper)/config/sim_params.yaml" command="load" />
  </node-->

  <node name="angle_finder" pkg="prop_mapper" type="angle_finder" output="screen" unless="$(arg use_mock_angle_finder)">
    <rosparam file="$(find prop_mapper)/config/sim_params.yaml" command="load" />
  </node>

  <node name="distance_finder" pkg="prop_mapper" type="distance_finder" output="screen">
    <rosparam file="$(find prop_mapper)/config/sim_params.yaml" command="load" />
    <remap from="/scan" to="/laser/scan"/>
  </node>

  <node name="coord_finder" pkg="prop_mapper" type="coord_finder" output="screen">
    <rosparam file="$(find prop_mapper)/config/sim_params.yaml" command="load" />
    <remap from="/mavros/local_position/pose" to="/gazebo_pose"/>
  </node>

  <node name="prop_map" pkg="prop_mapper" type="prop_map" output="screen">
    <rosparam file="$(find prop_mapper)/config/sim_params.yaml" command="load" />
  </node>

</launch>